src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/service/LocationService.kt

```diff
@@ -383,6 +383,40 @@ class LocationService(
     return locCapChange.toDto(includeParent = true, includeNonResidential = false)
   }
 
+  @Transactional
+  fun updateSpecialistCellTypes(id: UUID, specialistCellTypes: Set<SpecialistCellType>): LocationDTO {
+    val cell = cellLocationRepository.findById(id)
+      .orElseThrow { LocationNotFoundException(id.toString()) }
+
+    if (cell.isPermanentlyDeactivated()) {
+      throw ValidationException("Cannot change the specialist cell types of a permanently deactivated location")
+    }
+
+    // TODO check for empty sets when 0 working cap etc (e.g. validation rules)
```

Comment:
I assume you are coming back to this.  The validation can but pushed into the domain model too

Reply:
The rule is already there in code, 

https://github.com/ministryofjustice/hmpps-locations-inside-prison-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/locationsinsideprison/jpa/Cell.kt#L236-L238

Maybe pull this out to be called

Reply:
Thanks. I have pushed a new commit that should implement the logic.